daemon/algod/api/algod.oas2.json

```diff
@@ -1075,6 +1075,37 @@
         }
       }
     },
+    "/v2/ready": {
+      "get": {
+        "tags": [
+          "public",
+          "nonparticipating"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "schemes": [
+          "http"
+        ],
+        "summary": "Returns OK if healthy and fully caught up.",
```

Comment:
I'm not sure if this is 100% true...I think we would still return a 200 if we never used catchpoint catchup (using slow round by round catchup via the catchup service). In that case we would be "ready", but still on round 10 out of 24 million for example.

Reply:
Oh boy, I am not so familiar with "slow round" mode.

>using slow round by round catchup via the catchup service

I assume slow round by round catchup is on the opposite side of fast catchup. In that case, we should be fine, if we are way behind the latest round, right?

So what would happen, if we use this node to send some transaction, or do some other operations...?

Reply:
By "slow round" mode I just mean starting the node from genesis without using catchpoint catchup.

In this case the node operations should be fine in all the ways that justify a 200, but I wouldn't say the node is "fully caught up".

Reply:
I see, I was reading form the issue description from https://github.com/algorand/go-algorand/issues/4223 that:

> the node really isn't capable of accepting transactions, looking up blocks, or accounts or doing much of anything until its fully caught up.

So that is the motivation of enforcing 200 only if it is fully caught up, such that it is serving like a health endpoint, but a little more than that, as a readiness endpoint.

Reply:
right, for a regular round-by-round catchup, it will always 400 until your node reaches the latest round.

Fast catchup is similar, consider following 2 phases before a node reaches latest round:
1. still fast catchup against some catchpoint, then by `node.status` there exists a catchpoint, 400 err
2. catching up the residual rounds between catchpoint and latest round, still 400 error, sync time != 0.

Such that the handler's behavior holds.

Reply:
Perfect, thanks.  I'm testing locally now.
This'll be nice as I'll be able to scale the nodes trivially and not worry about them receiving requests before they're ready.